crates/primitives/src/alias.rs

```diff
@@ -2,93 +2,147 @@
 #[path = "tests/alias.rs"]
 mod tests;
 
-use std::fmt::{self};
+use std::fmt;
+use std::marker::PhantomData;
 use std::str::FromStr;
 
-use serde::{Deserialize, Serialize};
+use serde::{de, ser, Deserialize, Serialize};
 use thiserror::Error;
 
-#[derive(Clone, Debug, Eq, PartialEq, Serialize)]
-pub struct Alias(String);
+use crate::application::ApplicationId;
+use crate::context::ContextId;
+use crate::identity::PublicKey;
 
-#[derive(Clone, Debug, Error)]
-#[error("Invalid alias: {0}")]
-pub struct InvalidAlias(String);
+const MAX_LENGTH: usize = 50;
```

Comment:
Consider reducing this to 30, which is less than the minimum valid length for base58-encoded 32-byte hashes.

Reply:
Leaving this for now. Will revisit.